hibernate-core/src/main/java/org/hibernate/sql/results/spi/ListResultsConsumer.java

```diff
@@ -153,6 +153,7 @@ public List<R> consume(
 		final QueryOptions queryOptions = rowProcessingState.getQueryOptions();
 		RuntimeException ex = null;
 		try {
+			persistenceContext.beforeLoad();
```

Comment:
In 5.6 we also did that for scrollable results it seems. IMO we should add that there as well.
It might be best to do it in the same place as 5.6, which would roughly be in `org.hibernate.sql.exec.internal.JdbcSelectExecutorStandardImpl#executeQuery` and `org.hibernate.sql.exec.internal.JdbcSelectExecutorStandardImpl#executeQueryScroll` around the `doExecuteQuery` calls.

Reply:
not sure about putting the `beforeLoad()` in `org.hibernate.sql.exec.internal.JdbcSelectExecutorStandardImpl#executeQuery`. I put inside the ListResultConsumer so we are sure we will always call the `persistenceContext.afterLoad();`

Working on the scroll